nexus/src/app/sagas/instance_create.rs

```diff
@@ -479,53 +422,20 @@ async fn sic_create_default_primary_network_interface(
         .await
         .map_err(InsertNicError::into_external)
         .map_err(ActionError::action_failed)?;
-    Ok(())
-}
-
-async fn sic_create_network_interfaces_undo(
-    sagactx: ActionContext<SagaInstanceCreate>,
-) -> Result<(), anyhow::Error> {
-    // We issue a request to delete any interfaces associated with this instance.
-    // In the case we failed partway through allocating interfaces, we need to
-    // clean up any previously-created interface records from the database.
-    // Just delete every interface that exists, even if there are zero such
-    // records.
-    let osagactx = sagactx.user_data();
-    let datastore = osagactx.datastore();
-    let saga_params = sagactx.saga_params();
-    let opctx =
-        OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
-    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
-    let (.., authz_instance) = LookupPath::new(&opctx, &datastore)
-        .instance_id(instance_id)
-        .lookup_for(authz::Action::Modify)
-        .await
-        .map_err(ActionError::action_failed)?;
-    datastore
-        .instance_delete_all_network_interfaces(&opctx, &authz_instance)
-        .await
-        .map_err(ActionError::action_failed)?;
-    Ok(())
-}
-
-/// Create an ID for an external IP address for instance source NAT.
-async fn sic_allocate_instance_snat_ip_id(
-    _: ActionContext<SagaInstanceCreate>,
-) -> Result<Uuid, ActionError> {
-    Ok(Uuid::new_v4())
+    Ok(Some(interface_id))
 }
 
 /// Create an external IP address for instance source NAT.
 async fn sic_allocate_instance_snat_ip(
     sagactx: ActionContext<SagaInstanceCreate>,
-) -> Result<(), ActionError> {
+) -> Result<Uuid, ActionError> {
     let osagactx = sagactx.user_data();
     let datastore = osagactx.datastore();
     let saga_params = sagactx.saga_params();
     let opctx =
         OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
     let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
-    let ip_id = sagactx.lookup::<Uuid>("snat_ip_id")?;
+    let ip_id = Uuid::new_v4();
```

Comment:
There's a reason UUIDs are usually allocated in separate saga actions from "allocation functions" - if this saga action fails after `allocate_instance_snat_ip` runs, but before the "saga action completed" node is written to the saga log, we'd end up creating multiple snap IPs with distinct UUIDs.

Can we do this UUID allocation in a different saga action?